### ISSUE TITLE
Add static noise between stations

### ISSUE THREAD

**Issue by raphaelyancey** [raphaelyancey/Virtual_FM_Band#4]

**Comment by chunkysteveo** [comment]
...Ohhh... You working on this project again? Cool! I got as far as buying an 18's Sharpe radio cassette and some retro GTA stickers. Going to do a ViceCity version eventually!

**Comment by raphaelyancey** [comment]
@chunkysteveo Yes I've been fixing bugs lately (check out the commits) and been thinking on how to implement the noise :) I'd like it to be dynamic, so we can hear other stations mixed with synthesized noise.

**Comment by chunkysteveo** [comment]
That would be cool, some randomly generated white noise pattern or similar.

**Comment by raphaelyancey** [comment]
I implemented a static noise feature in the `ftr_noise` branch. Works by increasing noise tracks volume when any station volume is below a threshold. Will merge into `master` when I'll get the chance to test it on a Pi very soon.

Noise tracks are not included, but [I recorded](https://raw.githubusercontent.com/raphaelyancey/Virtual_FM_Band/ftr_noise/noise_1.mp3) [some](https://github.com/raphaelyancey/Virtual_FM_Band/blob/ftr_noise/noise_2.mp3?raw=true) on my synthesizer you can use! They must be placed in the `NOISE_PATH`.

**Comment by SebDominguez** [comment]
I only hear the noise when I'm at the end of my virtual band (not between my stations)

**Comment by raphaelyancey** [comment]
@SebDominguez That's most likely because the script thinks the noise files are regular stations.

- Are you on the good branch (`ftr_noise`)?
- Are your noise files in `$HOME/audio/noise`? (see [this line](https://github.com/raphaelyancey/Virtual_FM_Band/blob/ftr_noise/src/main.py#L38) — you can set `NOISE_PATH` in your `.env` file to override this)

**Comment by SebDominguez** [comment]
I think this is due to a bad installation on my part.
When I run the install script, it was stuck on the scipy install for more than 1 hour. I tried to install scipy after expanding my swap but still it seems that this. package won't compile on my Pi (or is just taking forever)

**Comment by raphaelyancey** [comment]
> it seems that this. package won't compile on my Pi (or is just taking forever)

It **is** taking forever, because `scipy` is a very big package and doesn't come precompiled for ARM as far as I know. On a Pi 3B+ it can take between half an hour and an hour.

The funny side is that the script is using only one fonction from `scipy` so it definitely can be optimized by rewriting this function and getting rid of `scipy`.

If you still experience the issue, please double-check that you are on `ftr_noise`. If not you should execute `cd /home/app && git checkout ftr_noise` & reboot or relaunch the script manually. That's because the default installation will install from `master` and this branch doesn't handle noise files yet.
